src/compute-client/src/service.rs

```diff
@@ -454,6 +468,37 @@ where
 
                 emit_response
             }
+            ComputeResponse::CopyToResponse(id, response) => {
+                // Incorporate new copy to responses; awaiting all responses.
+                let entry = self
+                    .pending_copy_tos
+                    .entry(id)
+                    .or_insert_with(Default::default);
+                let novel = entry.insert(shard_id, response);
+                assert!(novel.is_none(), "Duplicate copy to response");
+                // We may be ready to respond.
+                if entry.len() == self.parts {
+                    let mut response = CopyToResponse::Rows(0);
+                    for (_part, r) in std::mem::take(entry).into_iter() {
+                        response = match (response, r) {
+                            (_, CopyToResponse::Canceled) => CopyToResponse::Canceled,
+                            (CopyToResponse::Canceled, _) => CopyToResponse::Canceled,
+                            (_, CopyToResponse::Error(e)) => CopyToResponse::Error(e),
+                            (CopyToResponse::Error(e), _) => CopyToResponse::Error(e),
+                            (CopyToResponse::Rows(r1), CopyToResponse::Rows(r2)) => {
+                                // After the s3 sink has finished processing, each worker is
+                                // expected to send a `CopyToResponse` back, which we are aggregating
+                                // here.
```

Comment:
I would drop this comment. It's pretty obvious (at least if you are familiar with the code).

If you want to keep it, can you replace "worker" with "shard"? Shards are not always workers, they might also be processes in multi-process replicas.